src/test/groovy/edu/ie3/datamodel/io/source/csv/CsvSystemParticipantSourceTest.groovy

```diff
@@ -92,23 +92,39 @@ class CsvSystemParticipantSourceTest extends Specification implements CsvTestDat
     expect:
     def heatPumps = Try.of(() -> csvSystemParticipantSource.getHeatPumps(map(operators), nodeMap, map(types), map(thermalBuses)), SourceException)
 
-    if (heatPumps.success) {
-      heatPumps.data.get().size() == resultingSize
-      heatPumps.data.get() == resultingSet as Set
-    } else {
-      heatPumps.exception.get().class == SourceException
-    }
+    heatPumps.success
+    heatPumps.data.get().size() == resultingSize
+    heatPumps.data.get() == resultingSet as Set
 
     where:
     nodes               | operators               | types               | thermalBuses              || resultingSize | resultingSet
     [sptd.hpInput.node] | [sptd.hpInput.operator] | [sptd.hpInput.type] | [sptd.hpInput.thermalBus] || 1             | [sptd.hpInput]
+  }
+
+  def "A SystemParticipantSource with csv input should throw an exception from an invalid heat pump input file as expected"() {
+    given:
+    def csvSystemParticipantSource = new SystemParticipantSource(
+    Mock(TypeSource),
+    Mock(ThermalSource),
+    Mock(RawGridSource),
+    new CsvDataSource(csvSep, participantsFolderPath, fileNamingStrategy))
+    def nodeMap = map([sptd.hpInput.node])
+
+    expect:
+    def heatPumps = Try.of(() -> csvSystemParticipantSource.getHeatPumps(map(operators), nodeMap, map(types), map(thermalBuses)), SourceException)
+
+    heatPumps.failure
+    heatPumps.exception.get().class == SourceException
+
+    where:
+    nodes               | operators               | types               | thermalBuses              | resultingSize || resultingSet
     [sptd.hpInput.node] | []                      | [sptd.hpInput.type] | [sptd.hpInput.thermalBus] || 1             | [
       new HpInput(sptd.hpInput.uuid, sptd.hpInput.id, OperatorInput.NO_OPERATOR_ASSIGNED, sptd.hpInput.operationTime, sptd.hpInput.node, sptd.hpInput.thermalBus, sptd.hpInput.qCharacteristics, sptd.hpInput.type)
     ]
-    []                  | []                      | []                  | []                        || 0             | []
-    [sptd.hpInput.node] | []                      | []                  | []                        || 0             | []
-    [sptd.hpInput.node] | [sptd.hpInput.operator] | []                  | []                        || 0             | []
-    [sptd.hpInput.node] | [sptd.hpInput.operator] | [sptd.hpInput.type] | []                        || 0             | []
+    [] | [] | [] | []                                                        || 0             | []
+    [sptd.hpInput.node] | []                      | []                  | [] || 0             | []
+    [sptd.hpInput.node] | [sptd.hpInput.operator] | []                  | [] || 0             | []
+    [sptd.hpInput.node] | [sptd.hpInput.operator] | [sptd.hpInput.type] | [] || 0             | []
```

Comment:
Just a minor point, but the columns are now misaligned here. Please also check below